From 6a17a6ea5a52ad07ea205842604e9123fe93f3f0 Mon Sep 17 00:00:00 2001
From: Kevin Lannen <kevin@patroness.com>
Date: Wed, 6 Nov 2019 19:42:02 +0000
Subject: [PATCH] Add armv8 optimizations

---
 CMake/unix_config.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMake/unix_config.cmake b/CMake/unix_config.cmake
index e851fd4b4..6c670aa58 100644
--- a/CMake/unix_config.cmake
+++ b/CMake/unix_config.cmake
@@ -9,8 +9,8 @@ macro(os_set_flags)
         set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -mfpu=neon -mfloat-abi=hard -ftree-vectorize")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpu=neon -mfloat-abi=hard -ftree-vectorize")
     elseif(${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
-        set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -mstrict-align -ftree-vectorize")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mstrict-align -ftree-vectorize")
+        set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -O2 -mstrict-align -ftree-vectorize -march=armv8-a+simd+crypto+crc")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -mstrict-align -ftree-vectorize -march=armv8-a+simd+crypto+crc")
     else()
         set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -mssse3")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mssse3")
-- 
2.17.1

